### ISSUE TITLE
Test button color contrast

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyfive#492]
**Is your feature request related to a problem? Please describe.**

The button block and button element has CSS applied that changes the background color on hover. 
It mixes 85% of the background color with transparent: ```background-color: color-mix(in srgb, var(--wp--preset--color--contrast) 85%, transparent);```
But it leaves the text color unchanged.

The result is that the contrast ratio between the button background and the button text color changes depending on what the surrounding background color is.

The color change is not used as the only indicator that a button is being hovered.
But the contrast ratio between the button background color and the text color must be high enough for the text to still be readable.
For simplicity's sake let's assume that the text size is the default 18px and that the minimum contrast ratio is 4.5:1 (WCAG AA).

-------
The theme has 9 theme style variations including the default:

![Image alt="Styles sidebar with theme style variations visible"](https://github.com/user-attachments/assets/5a94b349-f1b7-4ae8-b9bf-66aa1a87f0f9)

Every style variation has 6 section styles:
![Image alt="The section style options are shown in the block settings of the group block" ](https://github.com/user-attachments/assets/365b70fb-beb4-4a6e-a371-5c95f45562af)

In short, these options are used to quickly change the colors of a group block.
How this works is that the users inserts one of the patterns, or, they add a group block themselves,
selects one of the 6 styles, and places a button inside the group.

What needs to be tested is that the button colors are still accessible when these section style options are used, in each theme style variation.
What makes this so time consuming is that I found that automated contrast checking tools can not check the color contrast when color-mix is used.

Example:

1. Go to Appearance > Editor. Select any template to edit.
2. Open the Styles sidebar and reset all saved style changes.
3. Insert a group block.
4. Insert two buttons: one default and one outlined. Remember to add the links to the buttons.
5. Test the color contrast.
6. Select the group block again. In the block settings sidebar, open the Styles tab.
7. In the Styles panel, select style 2 and test the color contrast. Repeat this for all 6 styles.

Repeat from step 2, for each theme style variation in the Styles sidebar > Browse styles.

**Comment by carolinan** [comment]
Default theme style variation
- [x]  All passes 🎉 In section style 2, the filled button has a text to background contrast ratio of 4.84:1 where 4.5:1 is the minimum.

Evening theme style variation
- [x] Button color passes 🎉

Noon theme style variation
- [x]  Button color passes 🎉

Twilight theme style variation
- [x]  Button color passes 🎉

Sunrise theme style variation
- [x]  Button color passes 🎉

Midnight theme style variation
- [x]  Button color passes 🎉

**Comment by troychaplin** [comment]
I took a moment to test out the styles and don't see any issues

![Image](https://github.com/user-attachments/assets/dcbc6643-45ab-4951-b693-86e7aff1ec6d)

![Image](https://github.com/user-attachments/assets/0bc9f205-00d6-4b23-ae44-964086c2493f)

**Comment by carolinan** [comment]
Afternoon theme style variation: **Failed**

- On hover, the default button outside the style sections, has a background color of `#65743E` and a text color of `#DAE7BD` (base)
This has a contrast ratio of 3.91:1.
- On hover, the default button in section style 1 has a contrast ratio of 3.75:1.
- On hover, the default button in section style 3 has a contrast ratio of 3.83:1

**Comment by carolinan** [comment]
Morning theme style variation: **Failed**

On hover, the default button in section style 4, has a contrast ratio of 4.18:1. 
![Image](https://github.com/user-attachments/assets/657b1e90-844b-4a83-86da-16dd1d516df0)

**Comment by carolinan** [comment]
@beafialho  In these two style variations listed above, the button color contrast is not working **on hover.**

**Comment by beafialho** [comment]
I'm not familiar with the technical approach taken for the hover colors and I find it hard to provide any helpful feedback here.

**Comment by beafialho** [comment]
Perhaps @jasmussen has some useful input?

**Comment by jasmussen** [comment]
I'll take a look after lunch, thank you for the ping!

**Comment by carolinan** [comment]
I think the palette colors may need to be updated.

Note that automated contrast tools are not able to find the correct background color, manual testing is needed.

**Comment by jasmussen** [comment]
Looking, please correct me if I'm missing nuance. It seems the first problem is this color style:

![Image](https://github.com/user-attachments/assets/b49af4d0-b27a-48df-9a6b-4fe947750348)

Top right corner button doesn't meet 4.5:1 text contrast between button background and text color. The way the hover styles are generated is through a color-mix function, so the button color is mixed with 15% transparency. Which means we can do two things:

1. We can update the Afternoon color that's used for the button color, or just the color of the button, to be darker. 
2. We can change the color-mix function of the Afternoon hover style to mix with black instead of transparent.

Here's 2:

![Image](https://github.com/user-attachments/assets/2a21fa0f-1235-481c-854b-f75d3fbc5dcd)

That's `color-mix(in srgb, var(--wp--preset--color--contrast) 85%, black)`.

Let me know which one seems simpler. 

Taking a look at Morning now.

**Comment by carolinan** [comment]
Though Morning, section style 4, could possibly be solved by updating the default button text color for this section only.

**Comment by jasmussen** [comment]
Yes, was about to say, on Morning, the main problem appears because the button text color is the same as the section style background color:

![Image](https://github.com/user-attachments/assets/6e252bc9-d041-4990-b729-dc63e80d8138)

If we just make that text color the black-ish color, contrast is fine:

![Image](https://github.com/user-attachments/assets/242140cd-eeb5-4b1d-9917-6810bf1838af)

**Comment by jasmussen** [comment]
For Afternoon, the alternative to updating the color mix function, is to use a darker base color for the buttons. Here's what that would look like, with Accent 3 instead of Contrast:

![Image](https://github.com/user-attachments/assets/2d727239-591b-4274-9217-426781b91391)

Sidenote, Contrast and Accent 4 appear the same:

![Image](https://github.com/user-attachments/assets/3e3e6073-0d03-44de-818b-ff92a09a51ad)

**Comment by carolinan** [comment]
Yeah there are several palettes with duplicate colors. I have brought this up a number of times but it seems to be intended.

**Comment by beafialho** [comment]
> We can update the Afternoon color that's used for the button color, or just the color of the button, to be darker.

This ⬆️ seems like a good alternative:

![Image](https://github.com/user-attachments/assets/5c96c8ed-113f-44c8-82f8-819780dc6fb2)
